tests/e2e/specs/auth-flow.test.js

```diff
@@ -0,0 +1,71 @@
+/* eslint-env node */
+/* global page, jestPuppeteer */
+
+/**
+ * WordPress dependencies
+ */
+import { activatePlugin, deactivatePlugin, createURL, visitAdminPage } from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import { deactivateAllOtherPlugins, resetSiteKit } from '../utils';
+
+const oauthClientConfig = JSON.stringify( {
+	'web': {
+		'client_id': 'test-client-id',
+		'client_secret': 'test-client-secret',
+		'project_id': 'test-project-id',
+		'auth_uri': 'https://accounts.google.com/o/oauth2/auth',
+		'token_uri': 'https://accounts.google.com/o/oauth2/token',
+		'auth_provider_x509_cert_url': 'https://www.googleapis.com/oauth2/v1/certs'
+	}
+} );
+
+function stubGoogleSignIn( request ) {
+	if ( request.url().startsWith( 'https://accounts.google.com/o/oauth2/auth' ) ) {
+		request.respond( {
+			status: 302,
+			headers: {
+				location: createURL( '/', 'oauth2callback=1&code=valid-test-code' )
+			}
+		} );
+	} else {
+		request.continue();
+	}
+}
+
+describe( 'Site Kit set up flow for the first time', () => {
+
+	beforeAll( async() => {
+		await deactivateAllOtherPlugins();
+		await resetSiteKit();
+		await activatePlugin( 'e2e-tests-oauth-callback-plugin' );
+	} );
+
+	afterAll( async() => {
+		await deactivateAllOtherPlugins();
+		await resetSiteKit();
```

Comment:
Resetting seems like a reasonable thing to do for every test. We could move this to the `bootstrap.js` file.

Reply:
Yeah, I agree. Should this be done as part of this PR (and change other existing tests) or should we have a dedicated issue for this refactoring?

Reply:
Let's add a separate issue for that. There seem to be a few possible ways to solve this, so it would be good to clarify in an implementation brief before diving right into it.

Reply:
Added https://github.com/google/site-kit-wp/issues/353